src/screens/SyncOnboarding/index.tsx

```diff
@@ -0,0 +1,226 @@
+import { from, of, Subscription, throwError, timer } from "rxjs";
+import {
+  mergeMap,
+  concatMap,
+  delayWhen,
+  map,
+  tap,
+  retryWhen,
+} from "rxjs/operators";
+import getVersion from "@ledgerhq/live-common/lib/hw/getVersion";
+import { withDevice } from "@ledgerhq/live-common/lib/hw/deviceAccess";
+import { TransportStatusError } from "@ledgerhq/errors";
+import { FirmwareInfo } from "@ledgerhq/live-common/lib/types/manager";
+import { ScreenName } from "../../const";
+import { SyncOnboardingStackParamList } from "../../components/RootNavigator/SyncOnboardingNavigator";
+
+// FIXME: Define an initial onboarding state - or cannot have an initial state in our use case ?
+const initialOnboardingState: OnboardingState = {
+  isOnboarded: false,
+  isInRecoveryMode: false,
+  isRecoveringSeed: false,
+  isConfirmingSeedWords: false,
+  seedPhraseType: "24-words",
+  currentSeedWordIndex: 0,
+};
+
+const pollingFrequencyMs = 1000;
+
+type Props = StackScreenProps<
+  SyncOnboardingStackParamList,
+  "SyncOnboardingWelcome"
+>;
+
+export const SyncOnboarding = ({ navigation, route }: Props): ReactElement => {
+  const { colors } = useTheme();
+  const [device, setDevice] = useState<Device | null>(null);
+  const [onboardingState, setOnboardingState] = useState<OnboardingState>(
+    initialOnboardingState,
+  );
+  const [stepIndex, setStepIndex] = useState<number>(0);
+
+  const { pairedDevice } = route.params;
+
+  // WIP: only for demo
+  const onboardingSteps = [
+    "Pairing device",
+    "Setting up pin",
+    `Writing seed words ${
+      !onboardingState.isConfirmingSeedWords &&
+      onboardingState.currentSeedWordIndex > 0
+        ? onboardingState.currentSeedWordIndex + 1
+        : ""
+    }`,
+    `Confirming seed words ${
+      onboardingState.isConfirmingSeedWords
+        ? onboardingState.currentSeedWordIndex + 1
+        : ""
+    }`,
+  ];
+
+  const handleOnPaired = useCallback((pairedDevice: Device) => {
+    console.log(
+      `SyncOnboarding: handleOnPaired ${JSON.stringify(pairedDevice)}`,
+    );
+
+    setDevice(pairedDevice);
+  }, []);
+
+  // Triggered when a new paired device is passed when navigating to this screen
+  // It avoids having a callback function passed to the PairDevices screen
+  useEffect(() => {
+    if (pairedDevice) {
+      handleOnPaired(pairedDevice);
+    }
+  }, [pairedDevice, handleOnPaired]);
+
+  // Polls device state to update the onboarding state
+  useEffect(() => {
+    let onboardingStatePollingSubscription: Subscription;
+
+    if (device) {
+      console.log(
+        `SyncOnboarding: 🧑‍💻 new device: ${JSON.stringify(device)}`,
+      );
+
+      onboardingStatePollingSubscription = timer(0, pollingFrequencyMs)
+        .pipe(
+          tap(i => {
+            console.log(`SyncOnboarding: ▶️ Polling ${i}`);
+          }),
+          concatMap(() =>
+            withDevice(device.deviceId)(t => from(getVersion(t))),
```

Comment:
Can't we simplify this greatly with 
```
withDevice(deviceId)((t => from(getVersion(t))).pipe(retryWhen(retryWhileErrors(acceptError)))
```
Where `acceptError` does something [like this](https://github.com/LedgerHQ/ledger-live-common/blob/master/src/hw/deviceAccess.ts#L161).